src/lib/swapper/swappers/ThorchainSwapper/endpoints.ts

```diff
@@ -146,9 +154,24 @@ export const thorchainApi: SwapperApi = {
         const publicClient = viemClientByChainId[chainId as EvmChainId]
         assert(publicClient !== undefined, `no public client found for chainId '${chainId}'`)
 
+        const expectedAmountOut = BigInt(longtailData?.longtailToL1ExpectedAmountOut ?? 0)
+        // Paranoia assertion - expectedAmountOut should never be 0 as it would likely lead to a loss of funds.
+        assert(
+          expectedAmountOut !== undefined && expectedAmountOut !== 0n,
+          'expectedAmountOut is undefined',
+        )
+
+        const amountOutMin = BigInt(
+          bnOrZero(expectedAmountOut.toString())
+            .times(bn(1).minus(slippageTolerancePercentageDecimal ?? 0))
+            .toFixed(0, BigNumber.ROUND_UP),
+        )
+
+        // Ensure we have this to prevent sandwich attacks on the first step of a LongtailToL1 trade.
+        assert(amountOutMin > 0n, 'expectedAmountOut is zero')
```

Comment:
Would there be much benefit in adding this check to the quote so it can return error monad with `TradeQuoteError.SellAmountBelowMinimum` ? If we return that error without a minimum amount the error will say
```
"Amount too low"
```

Reply:
https://github.com/shapeshift/web/pull/6052/commits/3c5c907403bda0f1fe7547a3eda5942c08de5ee3 👌 

<img width="442" alt="Screenshot 2024-01-23 at 11 45 46 am" src="https://github.com/shapeshift/web/assets/97164662/fc731f88-4962-4bd8-a31c-4f9532628daf">